### ISSUE TITLE
[bug] Can't Delete SCM attribute

### ISSUE THREAD

**Issue by cguldner** [conan-io/conan#7567]
There are certain cases where I want to be able to delete the `scm` attribute, but Conan really doesn't want me to.

* This gives `AttributeError: scm` on `delattr`:
```python
if hasattr(self, "scm"):
   delattr(self, "scm")
```

* Setting it to `None` doesn't allow me to upload it, fails in `conans/client/cmd/uploader.py`:

```python
if hasattr(conanfile, "scm") and (
    conanfile.scm.get("url") == "auto" or conanfile.scm.get("revision") == "auto"):
```

* Setting it to `{}` gives `ERROR: SCM not supported: None`

Can you allow us to delete it in one way or another?

### Environment Details (include every applicable attribute)
  * Conan version: 1.27.1
  * Python version: 3.5.2

**Comment by jgsogo** [comment]
Hi! Can I ask what you are trying to achieve? I'm sure we weren't thinking about dynamically adding/removing the `scm` attribute from the recipe class so, if it works, probably you should consider it UB and future Conan releases may change that behaviour.

It could work if you remove the attribute from the `init()` function, but please, let's see what is your use case and, if needed, we can think about alternatives (or implement the feature are you are requesting it).

Thanks!

**Comment by cguldner** [comment]
@jgsogo  All of the ways I tried deleting it were in the `init` function, so I would have expected at least one of them to work.

The use case is that I am trying to allow for building Conan in places where there is no internet access. Thus, I wouldn't want the source to be automatically checked out. This would be solved by verifying that `scm` isn't `None` before trying to access its properties [here](https://github.com/conan-io/conan/blob/release/1.28/conans/client/cmd/uploader.py#L297). This would also be solved by #7049.

Another use case for this is that I have a repo with submodules, and recipes for those submodules, and there are cases where I want to check out the source from git (like when running through CI/CD), but if I'm building locally, then I just want to use the source from the submodule

And another use case is when I want to do local builds without checking out the source from git. Our directory structure doesn't work with editable packages, because there is a bug as noted in #7417, and our Artifactory instance doesn't support package revisions, so I added a flag to our build script to copy over source from a local checkout. In that case, I don't want to waste time checking out source from git when I'm not going to use it anyway
